packages/validation/src/automaticEntityValidator.ts

```diff
@@ -36,14 +36,23 @@ export default class AutomaticEntityValidator<TTarget extends Record<string, any
   @observable isErrorsVisible: boolean;
 
   @observable errors: ValidationErrors<TTarget> = {};
+  @observable private _requiredProperties: string[] = [];
   private validatedProperties: string[] = [];
 
+  @computed
+  get requiredProperties() {
+    return this._requiredProperties;
+  }
+
   constructor(target: TTarget, entityValidationRules: IEntityValidationRules<TTarget>, isErrorsVisible: boolean) {
     this.isErrorsVisible = isErrorsVisible;
 
     for (const propertyName in entityValidationRules) {
       if (entityValidationRules.hasOwnProperty(propertyName)) {
         const rules = (entityValidationRules as Record<string, IPropertyValidationRules>)[propertyName];
+        if (rules.hasOwnProperty("required")) {
```

Comment:
To `required` bych rád nechal v nějakém configu a ideálně aby šlo specifikovat i více těch validačních pravidel

Reply:
Jo, tohle úzké provázání (včetně property `requiredProperties`) se mi moc nelíbí.

Nestačilo by si ty validation rules prostě uložit do public property a pak k tomu z `label` přistupovat jako `entity?.__validation?.validationRules?.[property]?.required` (tady by se hodila nějaká extension funkce `hasRequiredProperty(entity, property) => !!entity?.__validation?.validationRules?.[property]?.required`.

Já bych asi zkusil do nových validátorů přidat to, že při generování validačních pravidel, se taky vygeneruje něco jako tags/metadata se statickými poznámkami, které ke každé property může validátor přidat. Je to vlastně podobné, jako varianta výše s uložením originálních validation rules, až na to, že s validation rules musí konzument rozumět tomu, co které validační pravidlo znamená, kdežto s těmi metadaty může ten validátor výstupní data nějak předpřipravit.